flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -283,7 +285,7 @@ public String getShardIterator(KinesisStreamShard shard, String shardIteratorTyp
 	 * @param startShardId which shard to start with for this describe operation (earlier shard's infos will not appear in result)
 	 * @return the result of the describe stream operation
 	 */
-	private DescribeStreamResult describeStream(String streamName, String startShardId) throws InterruptedException {
+	private DescribeStreamResult describeStream(String streamName, @Nullable String startShardId) throws InterruptedException {
```

Comment:
Only Kinesis does. Kinesalite incorrectly ignores `startShardId`.
It's marked `Nullable` here, because on fresh startups this method will be called with a `null` as the start ID (on startup there will be no shard Id to start from).